backend/src/models/user.ts

```diff
@@ -59,87 +60,25 @@ export const User = mongoose.model('User', userSchema);
  * Authenticates a user by checking the provided email and password.
  * @param email - The email of the user to authenticate.
  * @param password - The password of the user to authenticate.
- * @param res - The Express response object for sending the authentication result.
- * @returns A JSON response indicating the success or failure of the authentication process.
+ * @returns If user signs in successfully, return username otherwise return null
  */
-export const authenticateUser = async (
-  email: string,
-  password: string,
-  res: Response
-) => {
+export const authenticateUser = async (email: string, password: string) => {
   try {
     const user = await User.findOne({ email });
-
     if (!user) {
-      return res.json({
-        success: false,
-        message: 'User not found'
-      });
+      return null;
     }
-
     const passwordMatch = await bcrypt.compare(password, user.password);
-
     // Check if the password matches
     if (!passwordMatch) {
-      return res.json({
-        success: false,
-        message: 'Invalid password'
-      });
+      return null;
     }
 
     // Successful login
-    const name = user.username;
-    const token = jwt.sign({ name }, 'bookwormctrlcsbookwormctrlcs', {
-      expiresIn: '1d'
-    });
-
-    res.cookie('token', token);
-    return res.json({
-      success: true,
-      message: 'Sign in successfully'
-    });
+    return user.username;
   } catch (error) {
-    // TODO: Not sure why this error is logged to console rather than returned to user. Needs testing
-
-    // eslint-disable-next-line no-console
     console.error('Error during authentication:', error);
```

Comment:
Minor critique. This will prevent ESLint from emitting warnings relating to this `console` statement
```suggestion
    // eslint-disable-next-line no-console
    console.error('Error during authentication:', error);
```